src/languages/scala.js

```diff
@@ -117,6 +117,16 @@ export default function(hljs) {
     contains: [ NAME ]
   };
 
+  const EXTENSION = {
+    className: 'keyword',
+    begin: /(?<=^\s*)extension(?=\s+[\[\(])/
+  };
+
+  const END = {
+    className: 'keyword',
+    begin: /(?<=^\s*)end(\s+extension|\b)/
```

Comment:
Is there some reason both these are special and can't simply be added to the keywords list?  Is there a reason they absolutely need mode rules instead?

Are these not reserved words everywhere?

Reply:
> Are these not reserved words everywhere?

Unfortunately, they are not reserved keywords. It is not part of the Scala 2 syntax and Scala 3 only allows them in this particular context. Otherwise, they are considered normal identifiers. We can have names `end`/`extension` that can be defined with `def extension: FileExtension = ...` or members called `end` such as in `line.end`.

> Is there some reason both these are special and can't simply be added to the keywords list? Is there a reason they absolutely need mode rules instead?

This is to avoid breaking code in Scala 2 and code in Scala 3 (in particular contexts).

Reply:
Ok, I figured once I saw you using the normal keyword feature in your other PR.  We'll still need to remove the look-behind here though and find another way.